flake.nix

```diff
@@ -273,6 +331,8 @@
           };
 
           LIBCLANG_PATH = "${pkgs.libclang.lib}/lib/";
+          ROCKSDB_LIB_DIR = "${rocksdb-7}/lib/";
+          LD_LIBRARY_PATH = "${rocksdb-7}/lib/";
```

Comment:
There's probably a better way to do this like `patchelf`ing the binary, but idk if that's practical here.

Reply:
After a bit of investigation: This is caused by us using `mold` which is not wrapped to inject `rpath`.

This downgrades it to `lld` and fixes the issue:

```
diff --git a/.cargo/config.toml b/.cargo/config.toml
index b9fefd1f9..c4e86492f 100644
--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -1,6 +1,6 @@
 [target.x86_64-unknown-linux-gnu]
 linker = "clang"
-rustflags = ["-C", "link-arg=-fuse-ld=mold"]
+rustflags = ["-C", "link-arg=-fuse-ld=lld"]
 
 # mold is currently broken on MacOS
 # [target.aarch64-apple-darwin]
diff --git a/flake.nix b/flake.nix
index 272e5c94e..f09497643 100644
--- a/flake.nix
+++ b/flake.nix
@@ -308,6 +308,7 @@
             openssl
             pkg-config
             perl
+            pkgs.llvmPackages.bintools
             rocksdb-7
           ] ++ lib.optionals stdenv.isDarwin [
             libiconv
@@ -332,7 +333,6 @@
 
           LIBCLANG_PATH = "${pkgs.libclang.lib}/lib/";
           ROCKSDB_LIB_DIR = "${rocksdb-7}/lib/";
-          LD_LIBRARY_PATH = "${rocksdb-7}/lib/";
           CI = "true";
           HOME = "/tmp";
         };
```

Article about it: https://matklad.github.io/2022/03/14/rpath-or-why-lld-doesnt-work-on-nixos.html